README.md

```diff
@@ -99,32 +98,50 @@ color you like:
 
 ## Directory tracking
 
+The `default-directory` would change when you change your working directory.
+and text in the vterm buffer has a text property `working-directory`.
+
+
+
+vterm enable `compilation-shell-minor-mode` by default.
+even you change your working directory, `next-error` `previous-error` still
+can find out the corresponding source code.
+
+
 For `zsh` put this in your `.zshrc`:
 
 ```zsh
-function chpwd() {
-    print -Pn "\e]51;A$(pwd)\e\\";
+vterm_prompt() {
+      print -Pn "\e]51;A$(pwd)\007"
 }
+PROMPT=$PROMPT'%{$(vterm_prompt)%}'
+
 ```
+please use `PROMPT`, and do NOT use `chpwd` or `preexec`,
+unless you don't want to using `compilation-shell-minor-mode`.
+The `working-directory` text property need be updated when a new shell prompt come out.
```

Comment:
Pleas use proper casing, e.g. `Please`, make the NOT bold instead of Uppercase. Maybe formulate like this:
```If you want to use `compilation-shell-minor-mode`, please use `PROMPT` and **not** `chpwd` or `preexec`.